### PR TITLE
Update password flag to be more idiomatic

### PR DESCRIPTION
**Pull request by chvck**

### CODE REVIEW

**Comment by chvck** [comment]
I don't love this but also cannot find another approach.

**Review by ingenthr — COMMENTED** [review]
So, the canonical way to do this in a CLI tool is to have two options:

Ask me for the password at stdin:
```
-p -
```

Use the string "foobar" as a password:
```
-p foobar
```

Then you can require -p as an argument and if it's elided, error out.  

Seems like it'd be a better approach.  Also, it's sometimes nice to be able to specify password is in a file.  This is a security consideration, since on UNIX users can see the command line flags from other users.

**Comment by ingenthr** [comment]
Related: https://www.netmeister.org/blog/passing-passwords.html

File: src/cli_options.rs [hunk 1]
```diff
@@ -58,9 +58,10 @@ impl Command for Cbsh {
                 "name to show in the shell",
                 None,
             )
-            .switch(
+            .named(
                 "password",
-                "use to specify a password to use for authentication",
+                SyntaxShape::String,
+                "use to trigger a prompt to specify a password to use for authentication",
```
Comment (ingenthr) [inline src/cli_options.rs:64]: I think with the change, this line isn't quite right, is it?  It's provide the password or prompt it `-` is specified.

**Review by ingenthr — CHANGES_REQUESTED** [review]
Note on the help there…

**Review by ingenthr — APPROVED** [review]

**Pull request merged**